### PR TITLE
libfs: add dir-related operations for the billy FS layer

### PR DESCRIPTION
**Pull request by strib**
Issue: KBFS-2338

### CODE REVIEW

File: libfs/fs.go [hunk 1]
```diff
@@ -276,7 +311,42 @@ func (fs *FS) ReadDir(p string) (fis []os.FileInfo, err error) {
...
 
 // MkdirAll implements the billy.Filesystem interface for FS.
-func (fs *FS) MkdirAll(filename string, perm os.FileMode) error {
+func (fs *FS) MkdirAll(filename string, perm os.FileMode) (err error) {
+	fs.log.CDebugf(fs.ctx, "MkdirAll %s", filename)
+	defer func() {
+		fs.deferLog.CDebugf(fs.ctx, "MkdirAll done: %+v", err)
+	}()
+
+	parts := strings.Split(filename, "/")
```
Comment (jzila) [inline libfs/fs.go]: `filepath.ToSlash(filename)` first

Comment (songgao) [inline libfs/fs.go]: hmm I was under the impression that it's gonna be `"/"` no matter the platform, like https://golang.org/pkg/net/http/#FileSystem . But I can't seem to find relevant doc for `billy`

Comment (jzila) [inline libfs/fs.go]: ¯\_(ツ)_/¯ I just wanted to make sure it was brought up.

Comment (songgao) [inline libfs/fs.go]: Yup it's a good point!

Comment (songgao) [inline libfs/fs.go]: Oh there's a `Filesystem.Join` [method](https://godoc.org/github.com/src-d/go-billy#Basic) so I guess that's up to us. But the comment on `Join` seems to suggest we should treat Windows and Unix differently.

**Pull request merged**